src/main/java/seedu/address/model/event/Deadline.java

```diff
@@ -0,0 +1,36 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import java.time.LocalDate;
+
+public class Deadline extends Event {
+    private final LocalDate taskDate;
+    private final String date;
+
+    public Deadline(String description, String date, boolean isDone) {
+        super(description, isDone);
+        requireNonNull(date);
+        this.taskDate = LocalDate.of(Integer.parseInt(date.split("-")[2]), Integer.parseInt(date.split("-")[1]),
+                Integer.parseInt(date.split("-")[0]));
+        this.date = date;
+    }
+
+    public LocalDate getTaskDate() {
+        return this.taskDate;
+    }
+
+    public String getDate() {
+        return this.date;
+    }
+
+    @Override
+    public Deadline markAsDone() {
+        return new Deadline(this.getDescription(), this.getDate(), true);
```

Comment:
Not sure about this, but if it is a private variable should we directly access it rather than using the public methods? It should reduce dependency between our methods, and avoid unintended changes.